src/rust/src/tests/ed25519.rs

```diff
@@ -0,0 +1,735 @@
+use crate::librustzcash_zebra_crypto_sign_verify_detached;
+
+fn hex_vk(s: &str) -> [u8; 32] {
+    let mut tmp = [0u8; 32];
+    let mut l = 0;
+    for (i, c) in s.chars().enumerate() {
+        if i % 2 == 0 {
+            l = c.to_digit(16).unwrap();
```

Comment:
Should there be some length checks on `s.chars()`? If the length is odd then we'll silently throw away data here. Also it seems like there should be a panic if the string provided is too long?

Reply:
:shrug: You're right, but these are just helpers for the test vectors so I don't think it's a big deal. Similarly, the `unwrap()` could be triggered on other kinds of bad input.

Reply:
This is definitely resolved as I've moved the tests to the C++ suite.